### PR TITLE
[#827] WebApp res id no longer stored in models db

### PR DESCRIPTION
**Pull request by shawncrawley**

### CODE REVIEW

**Comment by hydrobuild** [comment]
Can one of the admins verify this patch?

**Comment by shawncrawley** [comment]
@pkdash @hyi @selimnairb @ironfroggy So I was able to access the ToolResource object by doing a get query where resource.object_id=metadata.object_id like so:

`tool_res_obj = ToolResource.objects.get(object_id=res_type.object_id)`

It worked in all of the tests that I ran, but I am not sure why it works. Could one of you explain why it works? Is the object_id property a key that associates metadata objects with resource objects? What is the difference between the `metadata.id` and `metadata.object_id` properties? I originally tried the metadata.id correlation, but it fail in certain tests.

**Comment by hyi** [comment]
@shawncrawley yes, object_id should be used to query resource objects from metadata. object_id stores primary key values from the models you'll be relating to, as explained in detail here: https://docs.djangoproject.com/en/1.9/ref/contrib/contenttypes/

File: hs_core/page_processors.py [hunk 1]
```diff
@@ -57,17 +57,16 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
...
             if content_model_str in str(res_type.get_supported_res_types_str()).lower():
-                url_obj = res_type.content_object.url_bases.first()
-                tool_res_obj = hydroshare.get_resource_by_shortkey(url_obj.resShortID)
+                tool_res_obj = ToolResource.objects.get(object_id=res_type.object_id)
                 if tool_res_obj:
                     is_authorized = authorize(request, tool_res_obj.short_id, view=True, raises_exception=False)[1]
                     if is_authorized:
-                        tool_url = url_obj.value
+                        tool_url = tool_res_obj.content_object.url_bases.first().value
                         u = user.username if len(user.username) > 0 else "anonymous"
```
Comment (pkdash) [inline hs_core/page_processors.py]: Better to use ```user.is_authenticated()``` in place of ``` len(user.username) > 0```

File: hs_core/page_processors.py [hunk 2]
```diff
@@ -57,17 +57,16 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
...
         for res_type in SupportedResTypes.objects.all():
             if content_model_str in str(res_type.get_supported_res_types_str()).lower():
-                url_obj = res_type.content_object.url_bases.first()
-                tool_res_obj = hydroshare.get_resource_by_shortkey(url_obj.resShortID)
+                tool_res_obj = ToolResource.objects.get(object_id=res_type.object_id)
                 if tool_res_obj:
                     is_authorized = authorize(request, tool_res_obj.short_id, view=True, raises_exception=False)[1]
                     if is_authorized:
-                        tool_url = url_obj.value
+                        tool_url = tool_res_obj.content_object.url_bases.first().value
```
Comment (pkdash) [inline hs_core/page_processors.py]: Better to use ```metadata``` in place of ```content_object```
```tool_url = tool_res_obj.metadata.url_bases.first().value```

**Comment by shawncrawley** [comment]
@pkdash Suggested changes have been made.

**Comment by pkdash** [comment]
+1

**Pull request merged**